src/cox_utils.jl

```diff
@@ -0,0 +1,78 @@
+# Compute first and last
+firsts(s) = [s[t].status && (t==1 || s[t] > s[t-1]) for t = 1:length(s)]
```

Comment:
Looks like you could create generators rather than arrays here, that would avoid a few allocations (not sure it matters).